app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+class GroupsController < ApplicationController
+  include OrganizationAuthorization
+
+  before_action :ensure_team_management_flipper_is_enabled
+  before_action :set_group
+  before_action :set_grouping
+  before_action :set_member, only: [:add_membership, :remove_membership]
+
+  def add_membership
+    repo_access = RepoAccess.find_by(user: @user, organization: @organization)
+
+    if repo_access.present?
+      @group.repo_accesses << repo_access
+      render nothing: true, status: 204
+    else
+      render json: {
+        message: "User isn't a member of this classroom."
+      }, status: 422
+    end
+  end
+
+  def remove_membership
+    repo_access = RepoAccess.find_by(user: @user, organization: @organization)
+
+    if repo_access.present?
+      @group.repo_accesses.delete(repo_access)
+      render nothing: true, status: 204
+    else
+      render json: {
+        message: "User isn't a member of this classroom."
+      }, status: 422
+    end
+  end
+
+  private
+
+  def set_grouping
+    @grouping = Grouping.find_by!(slug: params[:grouping_id])
+  end
+
+  def set_group
+    @group = Group.includes(:repo_accesses).find_by!(slug: params[:id])
+  end
```

Comment:
Instead of `before_action` let's turn these into helper methods.

```ruby
def grouping
  @grouping ||= Grouping.find_by!(slug: params[:grouping_id])
end
helper_method :grouping

def group
  @group ||= Group.includes(:repo_accesses).find_by!(slug: params[:id])
end
helper_method :group
```